instance/instance_upgrade.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Helper functions to upgrade production instances
+
+Contents:
+    * DogwoodToEucalyptus1 - updates Dogwood to Eucalyptus.1 (major release)
+    * Eucalyptus1toEucalyptus2 - updates Eucalyptus.1 to Eucalyptus.2 (minor release)
+"""
+
+import logging
+
+from instance.models.openedx_instance import OpenEdXInstance
+from instance.tasks import spawn_appserver
+
+logger = logging.getLogger(__name__)
+
+
+class InstanceUpgrade(object):
+    INITIAL_RELEASE = None
+
+    def get_instances_to_upgrade(self):
+        return OpenEdXInstance.objects.filter(  # Select instances
+            active_appserver___status="running",  # that are running
+            openedx_release__contains=self.INITIAL_RELEASE,  # on {INITIAL_INSTANCE}
+            use_ephemeral_databases=False,  # and use persistent databases.
+        )
+
+    def do_upgrade(self, instances):
+        pass
```

Comment:
@e-kolpakov Nit: Is this here as a reminder to implement this method on child classes? If so, maybe make it `raise NotImplementedError`?